libass/ass.c

```diff
@@ -190,14 +190,23 @@ static int test_and_set_read_order_bit(ASS_Track *track, int id)
 {
     if (resize_read_order_bitmap(track, id) < 0)
         return -1;
-    int index = id / 32;
-    uint32_t bit = 1u << (id % 32);
+    int index = id >> 5; //div32
+    uint32_t bit = 1u << (id & 0x1F); //mod32
     if (track->parser_priv->read_order_bitmap[index] & bit)
         return 1;
     track->parser_priv->read_order_bitmap[index] |= bit;
     return 0;
 }
 
+static void clear_read_order_bit(ASS_Track *track, int id)
+{
+    int index = id >> 5;
+    if (index < track->parser_priv->read_order_elems) {
```

Comment:
Given `test_and_set_read_order_bimap` doesn't bother checking whether the id is in range, is there any reason why it’s needed here?

Reply:
Consider the following:
```c
ass_set_check_readorder(track, 0);

//Add some packets
while (pkt = pkt->next)
    ass_process_chunk(track, pkt->data, pkt->size, pkt->pts, pkt->dur);

ass_set_check_readorder(track, 1);
ass_prune_events(track, 0); //if index is not checked, memory corruption would ensue
```

Reply:
Thought changing `check_readorder` after init was forbidden, but apparently no such restriction exists. Makes sense then, thx